internal/proxy/fuse_test.go

```diff
@@ -110,13 +101,212 @@ func TestREADME(t *testing.T) {
 		t.Fatalf("expected README data, got no data (dir = %v)", dir)
 	}
 
-	if cErr := c.Close(); cErr != nil {
-		t.Fatalf("c.Close(): %v", cErr)
-	}
+	cleanup() // close the client
 
-	// verify that c.Close unmounts the FUSE server
+	// verify that the FUSE server is no longer mounted
 	_, err = ioutil.ReadFile(filepath.Join(dir, "README"))
 	if err == nil {
 		t.Fatal("expected ioutil.Readfile to fail, but it succeeded")
 	}
 }
+
+func tryDialUnix(t *testing.T, addr string) net.Conn {
+	var (
+		conn    net.Conn
+		dialErr error
+	)
+	for i := 0; i < 10; i++ {
+		conn, dialErr = net.Dial("unix", addr)
+		if conn != nil {
+			break
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	if dialErr != nil {
+		t.Fatalf("net.Dial(): %v", dialErr)
+	}
+	return conn
+}
+
+func TestFUSEDialInstance(t *testing.T) {
+	fuseDir := randTmpDir(t)
+	fuseTempDir := randTmpDir(t)
+	tcs := []struct {
+		desc         string
+		wantInstance string
+		socketPath   string
+		fuseTempDir  string
+	}{
+		{
+			desc:         "mysql connections create a Unix socket",
+			wantInstance: "proj:region:mysql",
+			socketPath:   filepath.Join(fuseDir, "proj:region:mysql"),
+			fuseTempDir:  fuseTempDir,
+		},
+		{
+			desc:         "postgres connections create a directory with a special file",
+			wantInstance: "proj:region:pg",
+			socketPath:   filepath.Join(fuseDir, "proj:region:pg", ".s.PGSQL.5432"),
+			fuseTempDir:  fuseTempDir,
+		},
+		{
+			desc:         "connecting creates intermediate temp directories",
```

Comment:
Should we have some failures cases in here?

What if we dial an instance that doesn't exist? what happens?

Reply:
I found failure cases below, closing this.